### PR TITLE
chore(arm): fix poll status bug

### PR DESCRIPTION
**Pull request by xzf0587**
1. update poll status to get sub-operations when resource type is deployments
2. add error telemetry for poll status

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 1]
```diff
@@ -228,23 +231,26 @@ export async function pollDeploymentStatus(deployCtx: DeployContext) {
...
-              const subOperations = await client.deploymentOperations.list(
-                operation.resourceGroupName,
-                operation.resourceName
-              );
-              subOperations.forEach((sub) => {
-                const subOperation = getRequiredOperation(sub, deployCtx);
-                if (subOperation) {
-                  currentStatus[subOperation.resourceName] = subOperation.status;
+              // get sub operations when resource type is deployments.
+              if (operation.resourceType === ConstantString.DeploymentResourceType) {
```
Comment (adashen) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:235]: if the resouceType is not deployment, we will not collect infomation?

Comment (xzf0587) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:233]: When it is not a deployment, it has no sub operation. But we will show it in output as well.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 2]
```diff
@@ -261,11 +265,20 @@ export async function pollDeploymentStatus(deployCtx: DeployContext) {
...
+        );
       }
-      deployCtx.logProvider?.warning(
-        `[${PluginDisplayName.Solution}] ${deployCtx.deploymentName} -> waiting to get deplomyment status [${tryCount}]`
+      const pollError = returnSystemError(
+        error,
+        SolutionSource,
+        SolutionError.FailedToPollArmDeploymentStatus
+      );
+      sendErrorTelemetryThenReturnError(
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:278]: we may just send the error telemetry when exceeds max retry count to reduce possible noises.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 3]
```diff
@@ -67,6 +67,8 @@ const bicepOrchestrationConfigFileName = "config.bicep";
 const templatesFolder = "./templates/azure";
 const configsFolder = `.${ConfigFolderName}/configs`;
 const parameterFileNameTemplate = `azure.parameters.${EnvNamePlaceholder}.json`;
+const pollWaitSeconds = 10;
+const MaxLogTimes = 4;
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:71]: `m`axLogTimes to make the variable name aligned
by the way, can we have a better name for it? not sure whether `maxRetryTimes` suits for it.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 2 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts]: It is fixed. Thanks.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 3 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts]: It is fixed. Thanks.

**Review by blackchoey — APPROVED** [review]

**Pull request merged**